Uchu.World/Objects/Components/Server/MissionGiverComponent.cs

```diff
@@ -111,6 +112,9 @@ public void HandleInteraction(Player player)
                                 // Allow them to start it
                                 break;
                             case MissionState.Active:
+                                // Display the in-progress mission instead of possibly continuing and offering a new mission.
+                                missionInventory.MessageOfferMission(playerMission.MissionId, GameObject);
+                                return;
```

Comment:
Doesn't this imply that once you start a mission you can't start other missions that appear in the chain until after this one is ready to be completed, or am I missing something?

Reply:
For that mission giver, yes. Because of how the code would work, it would iterate over the missions in the mission giver and give the last mission id that passed the condition. If it iterated over a valid Active mission and then a valid Available mission, it would set the mission to the Active, then the Available, and then give the player the Available one while the Acitve one is going (which causes problems on the Client since this isn't supported). This line makes it so if a mission is Active, it always displays the current mission and never gives a new mission.

Reply:
Alright, that sounds solid. I would just like to see this tested with mission givers that have multiple possible missions as well, for example Nexus Jay in NS, to make sure that those still function properly. Nexus Jay for example (after joining a faction) allows you to go on multiple mission paths to NT and GF.

Reply:
That specific case is working on the current branch. Was hoping that how it was cropped would make the file small enough to directly embed, but I was wrong.
https://imgur.com/Eqebr9c

Reply:
Awesome, thanks for making sure!